parsing/dialects/intermediate/dialects/dialect1/parserImpls.ftl

```diff
@@ -5741,3 +5740,124 @@ SqlSelectInto SqlSelectInto() :
             fromClause, whereClause);
     }
 }
+
+SqlCall FirstLastValue() :
+{
+    final SqlNode value;
+    final SqlNode over;
+    final List<SqlNode> args = new ArrayList<SqlNode>();
+    SqlNullTreatmentModifier nullTreatmentModifier = null;
+    final SqlAggFunction function;
+    final SqlCall firstLastCall;
+}
+{
+    (
+        <FIRST_VALUE> {
+            function = SqlStdOperatorTable.FIRST_VALUE;
+        }
+    |
+        <LAST_VALUE> {
+            function = SqlStdOperatorTable.LAST_VALUE;
+        }
+    )
+    <LPAREN>
+    value = CompoundIdentifier() { args.add(value); }
+    [
+        (
+            <IGNORE> {
+                nullTreatmentModifier = new SqlNullTreatmentModifier(getPos(),
+                    SqlKind.IGNORE_NULLS);
+            }
+        |
+            <RESPECT> {
+                nullTreatmentModifier = new SqlNullTreatmentModifier(getPos(),
```

Comment:
nit: Since the call to the SqlNullTreatmentModifier constructor is common between these two cases, suggest doing that after the clause ends. Instead, we can just set the SqlKind value in the respective cases.